src/mod_auth_pubtkt.c

```diff
@@ -439,6 +444,105 @@ static int cookie_match(void *result, const char *key, const char *cookie) {
 	return 1;
 }
 
+/* Look for a ticket in a header - maybe a cookie, maybe not */
+static char *get_header_ticket(request_rec *r) {
+	auth_pubtkt_dir_conf *conf = ap_get_module_config(r->per_dir_config, &auth_pubtkt_module);
+
+	char *ticket_decoded, *found_ticket;
+	const char *header_name, *header_list_str, *header_value;
+	size_t ticket_len;
+
+	header_list_str = (conf->auth_header_name) ? conf->auth_header_name : MOD_AUTH_PUBTKT_HEADER_NAME;
+
+	if( !*header_list_str ) {
+		ap_log_rerror(APLOG_MARK, APLOG_ERR, APR_SUCCESS, r,
+			"TKTAuthHeader directive is empty"
+		);
+		return NULL;
+	}
+
+	if (conf->debug >= 1) {
+		ap_log_rerror(APLOG_MARK, APLOG_DEBUG, APR_SUCCESS, r,
+			"TKT get_header_ticket: looking for ticket in '%s' header(s)",
+			header_list_str
+		);
+	}
+
+	while( *header_list_str && (header_name = ap_getword_conf(r->pool, &header_list_str)) ) {
```

Comment:
This loops through the directive options one by one, in order, and conditionalls tries to find a ticket using a cookie parsing or raw header method.